tests/conftest.py

```diff
@@ -58,6 +58,432 @@ def pytest_logger_logdirlink():
     return log_path
 
 
+@pytest.fixture()
+def rbd_secret_factory(request):
+    """
+    RBD secret factory. Calling this fixture creates new secret.
+    """
+    instances = []
+
+    def factory():
+        rbd_secret_obj = helpers.create_secret(
+            interface_type=constants.CEPHBLOCKPOOL
+        )
+        assert rbd_secret_obj, "Failed to create secret"
+        instances.append(rbd_secret_obj)
+        return rbd_secret_obj
+
+    def finalizer():
+        """
+        Delete the RBD secrets
+        """
+        for instance in instances:
+            if instance:
```

Comment:
```suggestion
            if not instance.is_deleted:
```
This `if instance` is unnecessary since we're iterating over `instances`, however we do now have the `is_deleted` property that @petr-balogh implemented. We should change these lines in each of the finalizers to check the value before calling delete.

Reply:
@fbalak any update on this change?

Reply:
It is part of a teardown of all edited fixtures.